### PR TITLE
ISIS autoconfig file copies multiple user's sample files

### PR DESCRIPTION
**Pull request by abuts**
Solves #15970 -- see details in the ticket description. 

ISIS direct inelastic users configuration script was copying sample reduction files for main experiment user only despite all experiments were added to the user profile. 

This is not-satisfactory for express experiments as no main user exist for this experiment, so after the changes, autoconfiguration script copies sample files for each experiment a user participates in. 

**To test:**
Test by code review. 

No much additional testing is necessary and actually possible as it relies and needs very specific computer configuration, provided by isiscompute. In addition to that, I've already deployed the script on isiscompute where it belongs and it works fine. The changes illustrated by two new unit tests, which verifies that for a user, participating in three experiments three configuration files were copied to three user' RB folders.

Changes are small and I hope obvious though clogged by pylint changes -- unfortunately can not do anything about it -- it does not let me further without decreasing pylint count.

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
- [x] Is the code of an acceptable quality?
- [x] Does the code conform to the coding standards? Is it well structured with small focussed classes/methods/functions?
- [x] Are there unit/system tests in place? Are the unit tests small and test the a class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?
##### Functional Tests
- [x] Do changes function as described? Add comments below that describe the tests performed?
- [x] How do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant documentation been added/updated?
- [ ] Is user-facing documentation written in a user-friendly manner?
- [ ] Has developer documentation been updated if required?
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by FedeMPouzols** [comment]
@abuts : I assume this doesn't need a release note. Let me know if you'd like to add something.

**Comment by abuts** [comment]
@FedeMPouzols :, thanks.  Its administrative ISISCompute script. I do not believe any release notes  are necessary due to very few people would be able to run it properly (ISISCOMPUTE administrators) and very specific application area (isiscompute itself). In addition to that, its a bugfix, made on request of the instrument scientists, and though they will be happy when it works nobody care about the details.

File: scripts/test/ISISDirecInelasticConfigTest.py [hunk 1]
```diff
@@ -69,21 +71,31 @@ def _set_up(self):
...
-        fh.write('#Test reduction file\n')
-        fh.write('Contents={0}'.format(contents))
-        fh.close()
-        return full_file
+            file_name = mcf._sample_reduction_file(instr_name)
+            all_files.append(file_name)
+            full_file = os.path.join(file_path,file_name)
+            if os.path.isfile(full_file):
+                os.remove(full_file)
+            fh=open(full_file,'w')
```
Comment (FedeMPouzols) [inline scripts/test/ISISDirecInelasticConfigTest.py]: For this type of code I find very convenient the scoping
`with open(...) as fh:`, which tends to be safer and saves you from having to do `fh.close()`.

**Comment by FedeMPouzols** [comment]
This PR is to update in the Mantid codebase inelastic scripts that are already deployed and used on the "isiscompute" cluster at ISIS.
- Changes the way files are looked for and copied to support so-called "express" experiments (handled by instrument scientists)
- Proper funcitonal tests are done in practice by the admins and users of the cluster, and cannot be replicated here. `ISISDirecInelasticConfigTest` has been updated.
- Code looks good. A pylint disable has been fixed. A "disable=too-many-instance-attributes" has been added which I think  is perfectly reasonable in its context.
- No release notes needed for this as it is a very local and specialized change.

:shipit:

File: scripts/test/ISISDirecInelasticConfigTest.py (hunk 1 above)
Comment (abuts) [inline scripts/test/ISISDirecInelasticConfigTest.py]: good point, thanks -- unfortunately can not recall the exact form each time I deal with such issue in Python. Will try to use it next time.

**Pull request merged**